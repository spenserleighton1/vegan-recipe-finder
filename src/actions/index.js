export const addRecipes = (recipes) => ({
  type: 'ADD_RECIPES',
  recipes
})

export const searchByIngredient = (ingredients) => ({
  type: 'SEARCH_BY_INGREDIENT', 
  ingredients
})