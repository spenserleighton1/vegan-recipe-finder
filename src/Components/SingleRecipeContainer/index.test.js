import React, { Component } from 'react';
import { shallow } from 'enzyme';
import { saveRecipe, deleteRecipe } from '../../actions'
import { SingleRecipeContainer, mapStateToProps, mapDispatchToProps } from './'

describe('SingleRecipeContainer', () => {
  it('should match the snapshot when it has a props object', () => {
    const recipe = { 
      title: 'some title',
      publisher: 'some publisher',
      ingredients: ['carrot', 'another carrot'],
      linkUrl: 'www.com',
      id: 666
    }
    const wrapper = shallow(<SingleRecipeContainer recipe={ recipe }/>)

    expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot when props object is empty', () => {
    const recipe = {}

    const wrapper = shallow(<SingleRecipeContainer recipe={recipe}/>)

    expect(wrapper).toMatchSnapshot()
  })

  it.skip('should call addFavorite when the button is clicked', () => {
    const recipe = { 
      title: 'some title',
      publisher: 'some publisher',
      ingredients: ['carrot', 'another carrot'],
      linkUrl: 'www.com',
      id: 666
    }
    const mockAddFavorite = jest.fn()
    const wrapper = shallow(<SingleRecipeContainer 
      recipe={ recipe } 
      addFavorite={ mockAddFavorite } />)


    wrapper.find('.favorite-btn').simulate('click')

    expect(mockAddFavorite).toHaveBeenCalled()
  })

  it.skip('should call saveRecipeID action when addFavorite is called', () => {
    const recipe = { 
      title: 'some title',
      publisher: 'some publisher',
      ingredients: ['carrot', 'another carrot'],
      linkUrl: 'www.com',
      id: 666
    }
    const mockAuthUser = { uid: 666 }
    const mockAction = jest.fn()
    const wrapper = shallow(<SingleRecipeContainer 
      recipe={ recipe } 
      saveRecipeID={ mockAction } 
      authUser={ mockAuthUser }/>)

    wrapper.instance().addFavorite()

    expect(mockAction).toHaveBeenCalled()
  })

  describe('mapDispatchToProps', () => {
    it('should return a props object with saveRecipeID', () => {
      const mockDispatch = jest.fn();

      const mappedProps = mapDispatchToProps(mockDispatch);
      const actionToDispatch = saveRecipe({id: 666})

      mappedProps.saveRecipeID({id: 666})
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('should return a props object with deleteRecipe', () => {
      const mockDispatch = jest.fn();

      const mappedProps = mapDispatchToProps(mockDispatch);
      const actionToDispatch = deleteRecipe({id: 666})

      mappedProps.deleteRecipe({id: 666})
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })

  describe('mapStateToProps', () => {
    it('should return a props object', () => {
      const mockState = { singleRecipe: 'sure'};
      const expected = { recipe: 'sure'};

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    })
  })

})