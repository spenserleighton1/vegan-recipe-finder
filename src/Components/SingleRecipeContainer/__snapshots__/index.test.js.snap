// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleRecipeContainer should match the snapshot when it has a props object 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SingleRecipeContainer
    recipe={
        Object {
            "id": 666,
            "ingredients": Array [
              "carrot",
              "another carrot",
            ],
            "linkUrl": "www.com",
            "publisher": "some publisher",
            "title": "some title",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <RecipeDetailsCard
        addFavorite={[Function]}
        authUser={undefined}
        id={666}
        ingredients={
                Array [
                        "carrot",
                        "another carrot",
                      ]
        }
        linkUrl="www.com"
        publisher="some publisher"
        title="some title"
/>,
      "className": "single-recipe",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "addFavorite": [Function],
        "authUser": undefined,
        "id": 666,
        "ingredients": Array [
          "carrot",
          "another carrot",
        ],
        "linkUrl": "www.com",
        "publisher": "some publisher",
        "title": "some title",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <RecipeDetailsCard
          addFavorite={[Function]}
          authUser={undefined}
          id={666}
          ingredients={
                    Array [
                              "carrot",
                              "another carrot",
                            ]
          }
          linkUrl="www.com"
          publisher="some publisher"
          title="some title"
/>,
        "className": "single-recipe",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "addFavorite": [Function],
          "authUser": undefined,
          "id": 666,
          "ingredients": Array [
            "carrot",
            "another carrot",
          ],
          "linkUrl": "www.com",
          "publisher": "some publisher",
          "title": "some title",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SingleRecipeContainer should match the snapshot when props object is empty 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SingleRecipeContainer
    recipe={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
